AK/HashTable.h

```diff
@@ -442,6 +442,24 @@ class HashTable {
         return removed_count;
     }
 
+    T pop()
+    {
+        VERIFY(!is_empty());
+        T element;
+        if constexpr (IsOrdered) {
+            element = *m_collection_data.tail.slot();
```

Comment:
Looks like this actually doesn't compile for ordered tables, since the bucket is a pointer:
```suggestion
            element = *m_collection_data.tail->slot();
```